File: code-examples/scala/usage-examples/deleteMany.scala
// Need to import necessary packages
import org.mongodb.scala._
import org.mongodb.scala.model.Filters.regex
import scala.concurrent.ExecutionContext.Implicits.global

// Replace with your MongoDB connection string
val mongoClient: MongoClient = MongoClient("<connection string uri>")

val database: MongoDatabase = mongoClient.getDatabase("sample_mflix")
val collection: MongoCollection[Document] = database.getCollection("movies")

// The filter to delete documents that have "Santa" in their title
val filter = regex("title", "Santa")

collection.deleteMany(filter).subscribe(new Observer[DeleteResult] {
  override def onNext(result: DeleteResult): Unit = println("Deleted " + result.getDeletedCount + " documents")
  override def onError(e: Throwable): Unit = e.printStackTrace()
  override def onComplete(): Unit = println("Operation Complete")
})

// Close the MongoDB connection
mongoClient.close()

File: code-examples/cpp/usage-examples/deleteMany.cpp
#include <bsoncxx/builder/stream/document.hpp>
#include <bsoncxx/json.hpp>
#include <mongocxx/client.hpp>
#include <mongocxx/instance.hpp>
#include <mongocxx/stdx.hpp>
#include <mongocxx/uri.hpp>

int main() {
    mongocxx::instance inst{};
    std::string uri = "<connection string uri>";
    mongocxx::client conn{mongocxx::uri{uri}};

    auto db = conn["sample_mflix"];
    auto collection = db["movies"];

    bsoncxx::builder::stream::document document{};
    document << "title" << bsoncxx::types::b_regex{"Santa"};

    auto result = collection.delete_many(document.view());
    std::cout << "Deleted " << result->deleted_count() << " documents" << std::endl;

    return 0;
}
